python/cudf/cudf/tests/test_concat.py

```diff
@@ -641,19 +640,35 @@ def test_concat_dataframe_with_multiIndex(df1, df2):
 @pytest.mark.parametrize("ignore_index", [True, False])
 @pytest.mark.parametrize("sort", [True, False])
 @pytest.mark.parametrize("join", ["inner", "outer"])
-def test_concat_join(objs, ignore_index, sort, join):
+@pytest.mark.parametrize("axis", [0, 1])
+@pytest.mark.xfail(
+    raises=NotImplementedError, reason="we do not support duplicate columns"
+)
```

Comment:
xfails are for when there's logical failures in the code as opposed to testing for expected behavior. Not supporting duplicate columns is a design choice of cuDF so throwing an exception in these situations is expected behavior and isn't "failing".

Instead of xfailing here can we use a `pytest.raises` in the test to assert we're raising the expected exception?

Reply:
Thanks for clarifying Keith! This is really helpful. I'll update this :D